doc/api/packages.md

```diff
@@ -313,6 +313,23 @@ module inside the subfolder. Any modules which are not public
 should be moved to another folder to retain the encapsulation
 benefits of exports.
 
+To support automatic file extensions with folder mappings, a substituation
```

Comment:
Can we add a warning that this will always append the extension, even if the path already has one? E.g. I wouldn't want people to expect that `pkg/features/x.json` would end up as anything but `pkg/features/x.json.js`.

Also, maybe something like this to distinguish this from the existing use of "automatic file extensions" where it's based on a list of possible extensions & file existence:

```suggestion
To support adding file extensions inside of folder mappings, a substituation
```

Reply:
[](https://github.com/nodejs/node-addon-api)